### DISCUSSION TITLE
对于加密的sqlite数据库是否支持

### DISCUSSION THREAD

**Discussion by NPUlrk** [dotnetcore/FreeSql#999] [Q&A] [answered]
对于加密的sqlite数据库是否支持？现在的连接字符串只能传入文件路径，通过SQLiteConnection创建的连接可以通过ChangePassword方法使用密码连接，请问freesql是否支持连接加密的数据库？

**Comment by luoyunchong** [comment, marked as answer]
.NET+Sqlite如何支持加密  https://www.cnblogs.com/igeekfan/p/net-sqlite-encryption.html

FreeSql.Provider.SqliteCore如何加密  https://www.cnblogs.com/igeekfan/p/freesql-sqlitecore-SQLCipher.html

**Reply by NPUlrk** [reply]
非常感谢，请问对于使用.net framework4.5 Winform程序是否有支持的计划？

**Reply by luoyunchong** [reply]
没有计划，上层依赖不支持。

SQLite 来源于公共领域 SQLite Is Public Domain、
确保代码不会受到任何专有或许可内容的污染，没有任何来自互联网上的未知来源复制。即全是原创的。

虽然是免费的，无需许可证，可用于任何目的，但如果你的公司必须要一个许可证，你也能申请授权https://sqlite.org/purchase/license.

但不支持加密。如果想支持登录加密，需要另外的扩展SQLite 加密扩展(SQLite Encryption Extension,)，具有读取/写入 AES 加密数据库的附加功能。具体授权可参考 https://www.sqlite.org/prosupport.html

**Reply by luoyunchong** [reply]
或者你可以去sqlite官网购买。.netstandard2.0支持.net framework4.6.1 


sqlcipher 付费：https://www.zetetic.net/sqlcipher/sqlcipher-ado/
sqlite 付费：https://sqlite.org/purchase/see
